app/javascript/components/tooltips/Concept.tsx

```diff
@@ -0,0 +1,244 @@
+ * Only 1 should be active at once, but given the chance of race conditions
+ * A array is used to track potentially multiple open
+ */
+
+const openTooltips: OpenTooltip[] = []
+
+const indexOfOpenTooltip = (id: string) =>
+  openTooltips.findIndex(([openId]) => openId === id)
+
+const hasOpenTooltip = (id: string) =>
+  indexOfOpenTooltip(id) === -1
+    ? openTooltips.length > 0
+    : openTooltips.length > 1
+
+const addOpenTooltip = (id: string, dispatcher: React.Dispatch<unknown>) => {
+  if (openTooltips.find(([openId]) => openId === id)) {
+    return
+  }
+
+  openTooltips.push([id, dispatcher])
+}
+
+const removeOpenTooltip = (id: string) => {
+  const index = indexOfOpenTooltip(id)
+  if (index !== -1) {
+    openTooltips.splice(index, 1)
+  }
+}
+
+const closeOtherOpenTooltips = (id: string) => {
+  openTooltips.forEach(([openId, openDispatcher]) => {
+    if (openId !== id) {
+      dispatch(openId, openDispatcher, 'force-close')
+    }
+  })
+}
+
+/**
+ * showReducer
+ * This serves as a state reducer for the Concept (tooltip) Component for use with
+ * the useReducer hook.
+ */
+function showReducer(
+  state: ShowState,
+  [id = '', dispatch, source, action]: [
+    string,
+    React.Dispatch<unknown>,
+    ShowActionSource,
+    ShowActionValue
+  ]
+) {
+  const nextState: ShowState = { ...state }
+  const parsedAction = action === 'show'
+  const nextShow = (state: ShowState): boolean =>
+    state.requestHover || state.requestFocus || state.hover || state.focus
+
+  switch (source) {
+    case 'request-hover':
+      nextState.requestHover = parsedAction && !hasOpenTooltip(id)
+      nextState.show = nextShow(nextState)
+      break
+    case 'request-focus':
+      nextState.requestFocus = parsedAction
+      nextState.show = nextShow(nextState)
+      break
+    case 'hover':
+      nextState.hover = parsedAction && !hasOpenTooltip(id)
+      nextState.show = nextShow(nextState)
+      break
+    case 'focus':
+      nextState.focus = parsedAction
+      nextState.show = nextShow(nextState)
+      break
+    case 'force-close':
+      nextState.show = false
+      break
+    default:
+      throw new Error('unhandled tooltip show action')
+  }
+
+  if (nextState.show) {
+    addOpenTooltip(id, dispatch)
+  } else {
+    removeOpenTooltip(id)
+  }
+
+  // request-focus and focus actions take precedence over mouse events
+  // so it can close any other for accessibility purposes
+  if (
+    (source === 'request-focus' || source === 'focus') &&
+    nextState.show &&
+    hasOpenTooltip(id)
+  ) {
+    closeOtherOpenTooltips(id)
+  }
+
+  return nextState
+}
+
+function dispatch(
+  id = '',
+  dispatcher: React.Dispatch<any>, // eslint-disable-line @typescript-eslint/no-explicit-any
+  source: ShowActionSource,
+  action?: ShowActionValue
+): void {
+  dispatcher([id, dispatcher, source, action])
+}
+
+export const Concept = ({
+  endpoint,
+  parent,
+  hoverRequestToShow,
+  focusRequestToShow,
+  parentSlug = '',
```

Comment:
The "parent" and the "slug" are only related here via some coincidence. This tooltip might reasonable appear on things that are not concepts. So this would be better loosely coupled from that, and named `conceptSlug`

```suggestion
  conceptSlug = '',
```